src/main/java/com/cloudbees/jenkins/support/impl/JenkinsLogs.java

```diff
@@ -112,9 +109,18 @@ protected void printTo(PrintWriter out) throws IOException {
                                     out.println("N/A");
                                 } else {
                                     try {
-                                        List<LogRecord> records = needHack
-                                                ? SlaveLogFetcher.getLogRecords(computer)
-                                                : computer.getLogRecords();
+                                        List<LogRecord> records = null;
+                                        if (needHack) {
+                                            if (computer.getChannel() != null) {
+                                                Future<List<LogRecord>> future = SlaveLogFetcher.getLogRecords(computer.getChannel());
+                                                records = future.get(REMOTE_OPERATION_TIMEOUT_MS, TimeUnit.MILLISECONDS);
+                                            }
+                                        }
+
+                                        if (records == null) {
```

Comment:
Correct. The initial intention with this line was to "guard" against the potential that channel _could_ be null in which case the log records won't exist. It will then try to make an attempt to grab the `computer` version of `getLogRecords` which from what I can see does not return null, but an empty list. If needed I can write the for-loop in the same way as the initial implementation.